report/renderer/details-renderer.js

```diff
@@ -35,7 +35,7 @@ const URL_PREFIXES = ['http://', 'https://', 'data:'];
 export class DetailsRenderer {
   /**
    * @param {DOM} dom
-   * @param {{fullPageScreenshot?: LH.Artifacts.FullPageScreenshot}} [options]
+   * @param {{fullPageScreenshot?: LH.Audit.Details.FullPageScreenshot}} [options]
```

Comment:
I think this starts to illustrate how the boundaries are hard to see here (glancing at this without context, I wouldn't expect `LH.Audit` to be within the self-contained report context

Long-term do you plan to renamespace the report to `LHReport` or `LH.Report.` or similar?

Reply:
> I think this starts to illustrate how the boundaries are hard to see here (glancing at this without context, I wouldn't expect `LH.Audit` to be within the self-contained report context
> 
> Long-term do you plan to renamespace the report to `LHReport` or `LH.Report.` or similar?

Yeah, it's a really good question. I'm not sure. I defaulted to maintaining the familiar `LH.*` hierarchy, but in the report files, the `LH.Audit` namespace is empty except for `Details`: 

```ts
module LH {
  module Audit {
     export import Details = AuditDetails;
  }
}
```
https://github.com/GoogleChrome/lighthouse/pull/12946/files#diff-2a34e64f4e5c4c358613feaff0fbc2da80d47238fb89ca37fdc07c96ce8e1c2cR22-R24

so seems doubly weird to maintain it? Not that useful in the report if considered in isolation, and if the goal is consistency across files, it's actually misleading because you might think e.g. `LH.Audit.ScoreDisplayMode`  is going to be in there too.

So long term it may make sense to re work the names, but then it's asking everyone to keep track of multiple names for the same thing...

Reply:
Maybe if https://github.com/microsoft/TypeScript/issues/41825 is fixed, core could keep the global types and the other areas with fewer files and fewer type needs could just do

```js
/** @typedef {import('../../lhr/audit-details')} AuditDetails */

export class DetailsRenderer {
  /**
   * @param {DOM} dom
   * @param {{fullPageScreenshot?: AuditDetails.FullPageScreenshot}} [options]
   */
  constructor(dom, options = {}) {
    this._dom = dom;
    this._fullPageScreenshot = options.fullPageScreenshot;
  }
}
```

Without that issue fixed, though, we'd be stuck with
```js
/** @typedef {import('../../lhr/audit-details').FullPageScreenshot} FullPageScreenshot */
/** @typedef {import('../../lhr/audit-details').OpportunityColumnHeading} OpportunityColumnHeading */
/** @typedef {import('../../lhr/audit-details').TableColumnHeading} TableColumnHeading */
/** @typedef {import('../../lhr/audit-details').List} List */
/** @typedef {import('../../lhr/audit-details').NodeValue} NodeValue */
// ...
```
:(